### ISSUE TITLE
JSON reporter emits wrong end column for multi-line spans

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#374]
# JSON reporter emits wrong end column for multi-line spans

- **Severity:** Medium (corrupts positions consumed by editor/CI integrations)
- **Status:** Verified by reproduction
- **Location:** `src/span.zig:193-199` (`LocationFormatter.jsonStringify`)

## The bug

```zig
// span.zig:193
const start_offset = self.span.span.start;
const len = self.span.span.len();
const start = findLineColumn(self.source, start_offset);
var end = findLineColumn(self.source[start_offset..], len);
end.line += start.line - 1;
end.column += start.column - 1;
```

The end position is computed relative to the span start and then shifted. The **line** adjustment is correct, but the **column** adjustment is only valid when the span ends on the same line it starts. If the span crosses a newline, the relative `end.column` is already an absolute column within its own line, and adding `start.column - 1` inflates it.

## User impact

Any tool consuming `--format json` (editor plugins, CI annotators) highlights the wrong end position for multi-line diagnostics — the end column is shifted right by `start.column - 1`.

## Reproduction

```sh
cat > cr2.zig <<'EOF'
fn bar() !u32 {
    return 1;
}
pub fn foo() !u32 {
    const x = bar() catch |e| {
        return
            e;
    };
    return x;
}
EOF
zlint --format json cr2.zig
# no-catch-return label:
#   {"start":{"line":6,"column":9},"end":{"line":7,"column":22}}
# actual end of `return\n            e` on line 7 is column 14, not 22.
```

(Inflation = `start.column - 1` = 8; 14 + 8 = 22.)

## Proposed fix

Only apply the column shift when the span is single-line:

```zig
var end = findLineColumn(self.source[start_offset..], len);
end.line += start.line - 1;
if (end.line == start.line) {
    end.column += start.column - 1;
}
```

(`findLineColumn` returns 1-based lines, so a relative `end.line` of 1 means "same line".) Add a unit test with a span crossing a newline.
